Update k3d/k3s (#633)
camptocamp/c2cciutils: README.md

            # and publish them with `c2cciutils-publish --group=<group>`
```

By default, the last line of the `SECURITY.md` file will be published (`docker`) with the tag
`latest`. Set `latest` to `False` to disable it.

With the `c2cciutils-clean` the images on Docker hub for `feature_branch` will be removed on branch removing.

## Download applications

> To manage application download with version managed by Renovate, you can:

Reviewer: Not sure to understand which types of applications are concerned.
IMHO it is important the user understand in introduction if this match with his use case.

As I understand this is only useful in CI, not in Docker builds, this is why you put files in ~/.local/bin (this cannot be the case with Docker as the user can change between runs).

I would propose :

```suggestion
In case some executables or applications from GitHub releases or any other URLs are required on the CI host and are not handled by any dependency manager, we provide a set of tools to install them and manage upgrades through Revovate.
```

Create an application file (e.-g. `applications.yaml`) with:

```yaml
# yaml-language-server: $schema=https://raw.githubusercontent.com/camptocamp/c2cciutils/master/c2cciutils/schema-applications.json

# Application from GitHub release
<organization>/<project>:
  get-file-name: <file name present in the release>
  to-file-name: <The file name you want to create in ~/.locale/bin>
  finish-command: # The command you want to run after the file is downloaded
    - - chmod # To be executable (usually required)
      - +x
      - <to-file-name>
    - - <to-file-name> # Print the version of the application
      - --version
# Application from GitHub release in a tar file (or tar.gz)
<organization>/<project>:
  get-file-name: <file name present in the release>
  type: tar
  tar-file-name: <The file name available in the tar file>

>   to-file-name: <The file name you want to create in ~/.locale/bin>

Reviewer: ```suggestion
  to-file-name: <The file name you want to create in ~/.local/bin>
```

  finish-command: [...] # The command you want to run after the file is downloaded
# Application from an URL
<application reference name>:
  url-pattern: <The URL used to download the application>

>   to-file-name: <The file name you want to create in ~/.locale/bin>

Reviewer: ```suggestion
  to-file-name: <The file name you want to create in ~/.local/bin>
```

  finish-command: [...] # The command you want to run after the file is downloaded
```

In the attributes `url-pattern`, `get-file-name` you can use the following variables:

- `{version}`: The version of the application present in the version file.
- `{version_quote}`: The URL encoded version.
- `{short_version}`: The version without the `v` prefix.

The `applications-versions.yaml` file is a map of applications and their versions.

Add in your Renovate configuration:

```json5

  regexManagers: [
    {
      fileMatch: ['^applications-versions.yaml$'],
      matchStrings: [
        '(?<depName>[^ ]+): (?<currentValue>[^ ]+) # (?<datasource>[^ ]+)',
      ],
    },
  ],
```

> Then you can call `c2cciutils-update-applications --applications-file=applications.yaml --versions-file=applications-version.yaml` to download or update the applications.

Reviewer: As I understand, we should call this in the CI before calling the considered application.

This does not update but only download applications (upgrades are proposed by Renovate):

`c2cciutils-update-applications` => `c2cciutils-download-applications`

I would propose:

```suggestion
Now you need to call `c2cciutils-download-applications --applications-file=applications.yaml --versions-file=applications-version.yaml` to install required applications on CI host before using them.
```

Reviewer: Note : `c2cciutils-update-applications` => `c2cciutils-download-applications`

Reviewer: Maybe add an example for GitHub Workflows
